rule/repository_memory.go

```diff
@@ -104,17 +121,18 @@ func (m *RepositoryMemory) Set(ctx context.Context, rules []Rule) error {
 	return nil
 }
 
-func (m *RepositoryMemory) Match(ctx context.Context, method string, u *url.URL) (*Rule, error) {
+func (m *RepositoryMemory) Match(_ context.Context, method string, u *url.URL) (*Rule, error) {
 	m.Lock()
 	defer m.Unlock()
 
 	var rules []Rule
 	for k := range m.rules {
 		r := &m.rules[k]
-		if err := r.IsMatching(method, u); err == nil {
+		if matched, err := r.IsMatching(m.matchingStrategy, method, u); err != nil {
+			return nil, errors.WithStack(err)
+		} else if matched {
 			rules = append(rules, *r)
 		}
-		m.rules[k] = *r
```

Comment:
This will disable caching of the matcher. I think that's ok for an interim solution but we should prevent recompiling the regular and glob expressions on every request. One thing to figure out is how that works depending on the matching strategy.